src/Dialect/ONNX/ONNXShapeHelper.hpp

```diff
@@ -63,6 +64,21 @@ struct ONNXSliceOpShapeHelper : public ONNXOpShapeHelper<ONNXSliceOp> {
   SmallVector<IndexExpr, 4> steps;
 };
 
+// Shape for GemmOp.
+struct ONNXGemmOpShapeHelper : public ONNXOpShapeHelper<ONNXGemmOp> {
+  ONNXGemmOpShapeHelper(
+      ONNXGemmOp *newOp, ConversionPatternRewriter *rewriter);
+
+  LogicalResult Compute(ONNXGemmOpAdaptor operandAdaptor);
+
+  // Additional data for GemmOp: output = a * b.
+  SmallVector<IndexExpr, 4> aDims; // Dim of A, after applying transpose.
+  SmallVector<IndexExpr, 4> bDims; // Dim of B, after applying transpose.
+  SmallVector<IndexExpr, 4> cDims; // Dim of C, padding "1" when broadcast.
+  bool hasBias; // Whether ther eis a bias (aka C exists).
+  int cRank; // Dim of the original C (not padding dims by 1).
```

Comment:
Something doesn't look right here. You're ending your block with an `int cRank` declaration.

Reply:
Actually it's fine I thought this was the inferShape method. :)

Reply:
I believe I need that init back. Have added it and I am running the backend tests.

Reply:
I added here params that I need both in the computation inside of shape helper as well as in the lowering. Instead of redundantly computing them, I just added them to the struct.